pkg/storage/util/util.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/operator-framework/rukpak/internal/storage"
+	"github.com/operator-framework/rukpak/internal/util"
+	pkgstorage "github.com/operator-framework/rukpak/pkg/storage"
+)
+
+const DefaultBundleCacheDir = "/var/cache/bundles"
+
+func NewBundleStorage(mgr ctrl.Manager, provisionerStorageDirectory, httpExternalAddr, bundlesPath string, opts ...HTTPOption) (pkgstorage.Storage, error) {
+	storageURL, err := url.Parse(fmt.Sprintf("%s/%s/", httpExternalAddr, bundlesPath))
+	if err != nil {
+		return nil, err
+	}
+	// URL.JoinPath needs go1.19 or later
+	// storageURL, err := url.Parse(httpExternalAddr)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// storageURL = storageURL.JoinPath("/", bundlesPath, "/")
```

Comment:
Was this a problem in the previous implementation too?

Reply:
No, it wasn't.  This is the suggested implementation in this PR but it can not be implemented unless we update the go to go1.19.